### ISSUE TITLE
SPAdes scaffolds assembly missing (skipping)

### ISSUE THREAD

**Issue by luke831215** [ablab/spades#45]
Hi all,

In the log file, I found **Skipping processing of scaffolds (empty file)** and the output therefore missed scaffolds.fasta but there was configs.fasta. Any clue if it is a memory issue or anything else?

I use veresion 3.11.1 and set num_thread to 50 and memory limit to 1024 GB and run on a machine with 96 cores and 1.5TB RAM. 

Any help is appreciated!

**Comment by asl** [comment]
Hello

Please provide us your spades.log file

**Comment by luke831215** [comment]
[params.txt](https://github.com/ablab/spades/files/1446379/params.txt)
[spades.log](https://github.com/ablab/spades/files/1446378/spades.log)

**Comment by asl** [comment]
This smells like an I/O issue. Does the problem reproduce if you'd simply restart everything from scratch?

**Comment by luke831215** [comment]
Yes, it seems so with the same input and parameters. According to our discussion, could it mean it is almost impossible to produce scaffolds from the assembled contigs? This way, we could simply treat it as a scaffold that contains only a single contig?

**Comment by asl** [comment]
> According to our discussion, could it mean it is almost impossible to produce scaffolds from the assembled contigs? 
Well, the situation with SPAdes is quite the opposite. contigs are generated from scaffolds, so the files are always there and contigs should be equal to scaffolds in this case. So what you're seeing is an indication of real issue.

Will it be possible for your to share your data with us so we could reproduce and fix the issue?

**Comment by luke831215** [comment]
Glad to share!

[params.txt](https://github.com/ablab/spades/files/1467789/params.txt)
[spades.log](https://github.com/ablab/spades/files/1467790/spades.log)
Dataset downloaded from SRA: fastq-dump --split-files SRR522246
Reference genome: https://goo.gl/8X3qCj
I use QUAST gage mode as benchmark tools.

**Comment by asl** [comment]
Your second spades.log clearly shows that the scaffolds.fasta are there and processed:

```
== Processing of scaffolds


== Running contig polishing tool: /home/luke831215/tools/SPAdes-3.11.1-Linux/bin/corrector /home/ph/mnt/biocollab/GAGE-B/b-sides/R_sphaeroides_MiSeq/assembly/mismatch_corrector/scaffolds/configs/corrector.info /home/ph/mnt/biocollab/GAGE-B/b-sides/R_sphaeroides_MiSeq/assembly/misc/assembled_scaffolds.fasta


== Dataset description file was created: /home/ph/mnt/biocollab/GAGE-B/b-sides/R_sphaeroides_MiSeq/assembly/mismatch_corrector/scaffolds/configs/corrector.info
```

Also, the second dataset is different from the first one. We need the one where the issue could be reproduced. We're certainly having the R.sphaeroides dataset from GAGE-B and everything is ok there.

**Comment by luke831215** [comment]
Sorry, this should be right.
[params.txt](https://github.com/ablab/spades/files/1469918/params.txt)
[spades.log](https://github.com/ablab/spades/files/1469919/spades.log)
Link for input reads: https://goo.gl/5wevh4

The reads here are the subset (93.4%) of R.sphaeroides MiSeq dataset. Please note that I don't input single reads using --pe1-s here. Adding up the single reads does produce scaffolds.fasta for this subset, but not in all other cases.

**Comment by asl** [comment]
> Sorry, this should be right.

Doesn't seem so. The log has:

```
== Processing of scaffolds


== Running contig polishing tool: /home/luke831215/tools/SPAdes-3.11.1-Linux/bin/corrector /home/ph/mnt/biocollab/GAGE-B/b-sides/R_sphaeroides_MiSeq/assembly/mismatch_corrector/scaffolds/configs/corrector.info /home/ph/mnt/biocollab/GAGE-B/b-sides/R_sphaeroides_MiSeq/assembly/misc/assembled_scaffolds.fasta


== Dataset description file was created: /home/ph/mnt/biocollab/GAGE-B/b-sides/R_sphaeroides_MiSeq/assembly/mismatch_corrector/scaffolds/configs/corrector.info
```

**Comment by luke831215** [comment]
![2017-11-14_163229](https://user-images.githubusercontent.com/6273619/32770130-6f106d48-c959-11e7-94fe-79468f71c287.png)

This is what I saw. Please check the new log file uploaded.

**Comment by ChQuinteroC** [comment]
Hi Luke
Did you resolve the problem? 
I have the exact problem, but only if i use --careful parameter

**Comment by Keyth-ru** [comment]
Hi guys, 
I have the same problem like ChQuinteroC I only use --careful

**Comment by ShobhitaShah** [comment]
I have installed spades, I am unable to produce the scaffolds while running either with "spades.py --test" nor with  spades.py -1 sample_1P.fastq.gz -2 sample_2P.fastq.gz -o spades_output. I am attaching params.txt and spades.log. Kindly help with solution.
[spades.log](https://github.com/ablab/spades/files/15480642/spades.log)
[params.txt](https://github.com/ablab/spades/files/15480643/params.txt)
